build/cmake/SetupBuildEnvironment.cmake

```diff
@@ -53,6 +53,9 @@ elseif(CC_IS_MINGW)
         set(CMAKE_EXE_LINKER_FLAGS "-Wl,--large-address-aware")
     endif (NOT BUILD_64)
 
+    add_definitions(-D_UNICODE)
+    add_definitions(-DUNICODE)
+
```

Comment:
Anodher appoarch is to make the use of `std::wstring` in src/framework/ui/internal/platform/windows/windowsplatformtheme.cpp dependant on whether `UNICODE` is `#define`'d or not:

    #if defined(UNICODE)
    static const std::wstring windowsThemeKey = L"Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize";
    static const std::wstring windowsThemeSubkey = L"AppsUseLightTheme";
    #else
    static const std::string windowsThemeKey = "Software\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize";
    static const std::string windowsThemeSubkey = "AppsUseLightTheme";
    #endif

Reply:
So this and the 'cleanup' change (dropping the `#if (defined (_MSCVER) || defined (_MSC_VER))...#else...#endif` stuff including the else branch) I proposed should fix the MinGW compile issue. No need to change whether to use UNICODE or not.

Reply:
@cbjeukendrup, your opinion (your code from ad8ad3f0c)?

And idea in the compiler warning?

Reply:
I see no reason for it, I believe that we can assume that UNICODE is always on by default.
If we need to turn it off somewhere, then turn it off only in these special places.

Reply:
It is on for MSVC, off for MinGW, and has been that way ever since @anatoly-os ported it to MSVC

Reply:
But if the decision is to require `UNICODE`for both fro now on, that should be fine.

Reply:
It does certainly simplify the code

Reply:
And we don't officially build with MinGW anyway, so this change would never affect anyone on any release